src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -710,6 +710,52 @@ TEST_CASE("txenvelope", "[tx][envelope]")
                         });
                     }
 
+                    SECTION("signature verification fails due to too many "
+                            "signatures (signed twice)")
+                    {
+                        auto tx = a1.tx({root.op(payment(a1, 100))});
+                        tx->getEnvelope().signatures.clear();
+
+                        // add signer twice.
+                        tx->addSignature(s1);
+                        tx->addSignature(s1);
+                        tx->getEnvelope().tx.seqNum++;
+                        a1.setSequenceNumber(a1.getLastSequenceNumber() - 1);
+                        auto setup = [&]() {
+                            SignerKey sk = alternative.createSigner(*tx);
+                            Signer sk1(sk, 1);
+                            a1.setOptions(setSigner(sk1));
+                            root.setOptions(setSigner(sk1));
+                            REQUIRE(getAccountSigners(a1, *app).size() == 2);
+                            REQUIRE(getAccountSigners(root, *app).size() == 1);
+                            alternative.sign(*tx);
+                        };
+                        for_versions({3, 4, 5, 6, 8}, *app, [&] {
```

Comment:
I think it's sufficient to just test how this worked in versions 9 and 10, because that is when the behavior changed. This demonstrates that the invariant may once have been true, but definitely is no longer.